src/main/drivers/motor.c

```diff
@@ -138,6 +138,14 @@ static void analogInitEndpoints(const motorConfig_t *motorConfig, float outputLi
     }
 }
 
+bool isProtocolBidirectionalDshot(const motorDevConfig_t *motorDevConfig)
+{
+    bool motorProtocolDshot = false;
+    const bool motorProtocolEnabled = checkMotorProtocolEnabled(motorDevConfig, &motorProtocolDshot);
```

Comment:
I looked into this more... talk about complicated.
in config.c:
```
    if ((!configuredMotorProtocolDshot || (motorConfig()->dev.useDshotBitbang == DSHOT_BITBANG_OFF && (motorConfig()->dev.useBurstDshot == DSHOT_DMAR_ON || nChannelTimerUsed))) && motorConfig()->dev.useDshotTelemetry) {
        motorConfigMutable()->dev.useDshotTelemetry = false;
    }
```
Wouldn't that mean that `dev.useDshotTelemetry` would be true only if the motor protocol was DShot, and telemetry was enabled?
And hence could we use that?

Reply:
I do see a whole host of different definitions and functions relating to whether DShot Telemetry is active or requested.  Surely that code can be refactored more simply and efficiently?

Reply:
Yes, it looks like it gets fixed there in case something is configured wrong. But it is still cleaner to check this like limon is doing it. Otherwise this is dependent on the config clean-up code in `validateAndFixConfig()`.